src/charm.py

```diff
@@ -72,13 +78,42 @@ def main(self, event):
 
         self.model.unit.status = ActiveStatus()
 
+    def _check_dex_oidc_config_relation(self) -> None:
+        """Check for exceptions from the library and raises ErrorWithStatus to set the unit status.
+
+        Raises:
+            ErrorWithStatus: if the relation hasn't been established, set unit to BlockedStatus
+            ErrorWithStatus: if the relation has empty or missing data, set unit to WaitingStatus
+        """
+        try:
+            self._dex_oidc_config_requirer.get_data()
+        except DexOidcConfigRelationMissingError as rel_error:
+            raise ErrorWithStatus(
+                f"{rel_error.message} Please add the missing relation.", BlockedStatus
+            )
+        except DexOidcConfigRelationDataMissingError as data_error:
+            self.logger.error(f"Empty or missing data. Got: {data_error.message}")
+            raise ErrorWithStatus(
+                f"Empty or missing data in {OIDC_PROVIDER_INFO_RELATION} relation."
+                " This may be transient, but if it persists it is likely an error.",
+                WaitingStatus,
+            )
+
+    @property
+    def _dex_oidc_config_requirer(self) -> DexOidcConfigRequirer:
+        """Return a DexOidcConfigRequirer object that gathers data from an OIDC provider."""
+        return DexOidcConfigRequirer(
+            charm=self,
+            relation_name=OIDC_PROVIDER_INFO_RELATION,
+        )
+
     @property
     def service_environment(self):
         """Return environment variables based on model configuration."""
         secret_key = self._check_secret()
         skip_urls = self.model.config["skip-auth-urls"] or ""
         dex_skip_urls = "/dex/" if not skip_urls else "/dex/," + skip_urls
-
+        oidc_provider = self._dex_oidc_config_requirer.get_data().issuer_url
```

Comment:
Use the already stored data here, in other words, instantiate this in main during `_check_dex_oidc_config_relation()`.